File: apps/ios/Reels/Features/Picker/PickerModels.swift
import Foundation

// MARK: - Picker Models

struct PickerPlanSummary: Codable, Identifiable {
    let id: String
    let filmTitle: String
    let filmPosterPath: String?
    let status: String
    let participantCount: Int
    let createdAt: String
}

struct PickerPlanDetail: Codable {
    let id: String
    let filmTitle: String
    let filmTmdbId: Int?
    let filmPosterPath: String?
    let filmYear: Int?
    let pathway: String
    let status: String
    let organizer: Organizer
    let showtimes: [PickerShowtime]
    let participants: [PickerParticipantInfo]
    let confirmedShowtime: PickerShowtime?
    let expiresAt: String
    let createdAt: String
    let currentParticipantId: String?

    struct Organizer: Codable {
        let id: String
        let name: String
    }
}

struct PickerShowtime: Codable, Identifiable {
    let id: String
    let cinemaName: String
    let cinemaCity: String
    let date: String
    let time: String
    let ticketUrl: String?
    let voteCount: VoteCount?

    struct VoteCount: Codable {
        let available: Int
        let unavailable: Int
        let maybe: Int
    }
}

struct PickerParticipantInfo: Codable, Identifiable {
    let id: String
    let displayName: String
    let isOrganizer: Bool
}

struct FilmSearchResult: Codable, Identifiable {
    let tmdbId: Int
    let title: String
    let year: Int?
    let posterPath: String?
    let overview: String?

    var id: Int { tmdbId }
}

struct CreatePlanInput: Encodable {
    let filmTitle: String
    let filmTmdbId: Int?
    let filmPosterPath: String?
    let filmYear: Int?
    let pathway: String
    let city: String?
    let cinema: String?
    let targetDate: String?
    let showtimes: [ShowtimeInput]

    struct ShowtimeInput: Encodable {
        let cinemaName: String
        let cinemaCity: String
        let date: String
        let time: String
        let ticketUrl: String?
        let isManualEntry: Bool
    }
}

struct CreatePlanResponse: Codable {
    let planId: String
    let shareUrl: String
    let expiresAt: String
}

struct JoinPlanInput: Encodable {
    let planId: String
    let displayName: String
    let guestSessionToken: String?
}

struct JoinPlanResponse: Codable {
    let participantId: String
    let sessionToken: String?
}

struct VoteInput: Encodable {
    let participantId: String
    let votes: [VoteEntry]

    struct VoteEntry: Encodable {
        let showtimeId: String
        let status: String
    }
}

struct VoteResponse: Codable {
    let success: Bool
    let updatedCount: Int
}

struct ConfirmInput: Encodable {
    let planId: String
    let showtimeId: String
}

struct ConfirmResponse: Codable {
    let success: Bool
    let confirmedShowtime: PickerShowtime
}

struct MyPlansResponse: Codable {
    let plans: [PickerPlanSummary]
}

struct SearchFilmsResponse: Codable {
    let results: [FilmSearchResult]
}

struct ShowtimesResponse: Codable {
    let showtimes: [PickerShowtime]
    let source: String
}

File: apps/android/app/src/main/java/com/reels/app/features/picker/PickerModels.kt
package com.reels.app.features.picker

import kotlinx.serialization.Serializable

@Serializable
data class PickerPlanSummary(
    val id: String,
    val filmTitle: String,
    val filmPosterPath: String? = null,
    val status: String,
    val participantCount: Int,
    val createdAt: String,
)

@Serializable
data class PickerPlanDetail(
    val id: String,
    val filmTitle: String,
    val filmTmdbId: Int? = null,
    val filmPosterPath: String? = null,
    val filmYear: Int? = null,
    val pathway: String,
    val status: String,
    val organizer: Organizer,
    val showtimes: List<PickerShowtime>,
    val participants: List<PickerParticipantInfo>,
    val confirmedShowtime: PickerShowtime? = null,
    val expiresAt: String,
    val createdAt: String,
    val currentParticipantId: String? = null,
) {
    @Serializable
    data class Organizer(val id: String, val name: String)
}

@Serializable
data class PickerShowtime(
    val id: String,
    val cinemaName: String,
    val cinemaCity: String,
    val date: String,
    val time: String,
    val ticketUrl: String? = null,
    val voteCount: VoteCount? = null,
) {
    @Serializable
    data class VoteCount(val available: Int, val unavailable: Int, val maybe: Int)
}

@Serializable
data class PickerParticipantInfo(
    val id: String,
    val displayName: String,
    val isOrganizer: Boolean,
)

@Serializable
data class FilmSearchResult(
    val tmdbId: Int,
    val title: String,
    val year: Int? = null,
    val posterPath: String? = null,
    val overview: String? = null,
)

@Serializable
data class CreatePlanRequest(
    val filmTitle: String,
    val filmTmdbId: Int? = null,
    val filmPosterPath: String? = null,
    val filmYear: Int? = null,
    val pathway: String,
    val city: String? = null,
    val cinema: String? = null,
    val targetDate: String? = null,
    val showtimes: List<ShowtimeInput>,
) {
    @Serializable
    data class ShowtimeInput(
        val cinemaName: String,
        val cinemaCity: String,
        val date: String,
        val time: String,
        val ticketUrl: String? = null,
        val isManualEntry: Boolean = false,
    )
}

@Serializable
data class CreatePlanResponse(val planId: String, val shareUrl: String, val expiresAt: String)

@Serializable
data class JoinPlanRequest(val planId: String, val displayName: String, val guestSessionToken: String? = null)

@Serializable
data class JoinPlanResponse(val participantId: String, val sessionToken: String? = null)

@Serializable
data class VoteRequest(val participantId: String, val votes: List<VoteEntry>) {
    @Serializable
    data class VoteEntry(val showtimeId: String, val status: String)
}

@Serializable
data class VoteResponse(val success: Boolean, val updatedCount: Int)

@Serializable
data class ConfirmRequest(val planId: String, val showtimeId: String)

@Serializable
data class ConfirmResponse(val success: Boolean, val confirmedShowtime: PickerShowtime)

@Serializable
data class MyPlansResponse(val plans: List<PickerPlanSummary>)

@Serializable
data class SearchFilmsResponse(val results: List<FilmSearchResult>)

@Serializable
data class ShowtimesResponse(val showtimes: List<PickerShowtime>, val source: String)
